earth_enterprise/src/scons/getversion.py

```diff
@@ -42,16 +43,125 @@ def GetLongVersion(backupFile, label=''):
+    rawDescription = _GetCommitRawDescription()
+    if _IsCurrentCommitTagged(rawDescription):
+        # Extract version name and build number
+        # from the tag (should be a release build tag)
+        splitTag = rawDescription.split('-')
+        return splitTag[0], splitTag[1]
     else:
-        return _VersionFromTagHistory(raw)
+        # Use branch name if we are not a detached HEAD
+        branchName = _GitBranchName()
+        if not branchName:
+            # we are a detached head not on a tag so just treat the
+            # raw describe like a topic branch name
+            return rawDescription, str(_GitCommitCount())
+        else:
+            # Get the version name from the branch name
+            if _IsReleaseBranch(branchName):
+                versionName = _GetReleaseVersionName(branchName)
+                return versionName, '{0}.{1}'.format(_GitBranchedBuildNumber(versionName), _GitCommitCount('HEAD', versionName))
+            else:
+                return _sanitizeBranchName(branchName),  str(_GitCommitCount())
+
+
+def _GitBranchedBuildNumber(versionName):
+    """Returns what the build number was from the branch point"""
+    prevRelTag = _GitPreviousReleaseTag(versionName)
+    prevBuildNumber = ''
+
+    if prevRelTag:
+        prevBuildNumber = prevRelTag.split('-')[1]
+    else:
+        prevBuildNumber = str(_GitCommitCount(versionName))
+
+    return prevBuildNumber
+
+
+def _GitPreviousReleaseTag(versionName):
+    """Looks for the tail tag for this versionName (the tag with the same name as the version name)
+    and if it finds it then it looks for any release build tags that are also pointing to the same
+    commit"""
+    tags = _GetRepository().tags
+    tailTag = next((tag for tag in _GetRepository().tags if tag.name == versionName), None)
+    if tailTag is None:
+        return ''
+
+    for tag in tags:
+        if tag.commit == tailTag.commit and tag.name <> tailTag.name and _IsReleseBuildTag(tag.name):
+            return tag.name
+    
+    return ''
+
+
+def _IsReleseBuildTag(tagName):
+    return re.match(r'[0-9]*\.[0-9]*\.[0-9]*\-*', tagName)
+
+
+def _IsReleaseBranch(branchName):
+    """Check if the branch name is a release branch"""
+    # a release branch begins with 'release_' and has
+    # a base tag that matches the release name
+    if branchName[:8] == 'release_':
+        versionName = _GetReleaseVersionName(branchName)
+        if _gitHasTag(versionName):
+            return True
+        else:
+            # see if we can pull the tag down from any of the remotes
+            repo = _GetRepository()
+            for remote in repo.remotes:
+                try:
+                    remote.fetch('+refs/tags/{0}:refs/tags/{0}'.format(versionName), None, **{'no-tags':True})
+                except git.exc.GitCommandError:
+                    pass
+            
+            # try one more time after the fetch attempt(s)
+            return _gitHasTag(versionName)
+
```

Comment:
yes I changed it to `return (_gitHasTag(versionName) is not None)` which will return a Boolean now instead of an object which might be None.  This is the else case here.

Reply:
@tst-rwildes I think there's still an implicit return None if the first conditional, if branchName[:8] == 'release_':, is false.